src/routes/movies.py

```
@@ -1,13 +1,194 @@
+        }
+    }
+)
+async def movie_detail(
+        movie_id: int,
+        db: AsyncSession = Depends(get_db),
+) -> MovieCreateResponseSchema:
+    movie = await get_movie_or_404(movie_id, db)
+    return MovieCreateResponseSchema.model_validate(movie)
+
+
+@router.delete(
+    "/{movie_id}/",
+    status_code=204,
+    responses={
+        404: {
+            "model": MovieErrorSchema,
+            "description": "Movie with the given ID was not found."
+        }
+    }
+)
+async def delete_movie(
+        movie_id: int,
+        db: AsyncSession = Depends(get_db),
+) -> None:
+    movie = await get_movie_or_404(movie_id, db)
+    await db.delete(movie)
+    await db.commit()
+
+
+@router.patch(
+    "/{movie_id}/",
+    response_model=MovieUpdateResponseSchema,
+    responses={
+        400: {
+            "model": MovieErrorSchema,
+            "description": "Invalid input data."
+        },
+        404: {
+            "model": MovieErrorSchema,
+            "description": "Movie with the given ID was not found."
+        }
+    }
+)
+async def update_movie(
+        movie_id: int,
+        movie_data: MovieUpdateRequestSchema,
+        db: AsyncSession = Depends(get_db),
+) -> MovieUpdateResponseSchema:
+
+    movie = await get_movie_or_404(movie_id, db)
+    data_dict = movie_data.model_dump(exclude_unset=True)
 
+    if (
+        (movie.date is not None and movie.date > date.today() + timedelta(days=365))
+        or (movie.score is not None and (movie.score < 0 or movie.score > 100))
+        or (movie.budget is not None and movie.budget < 0)
+        or (movie.revenue is not None and movie.revenue < 0)
+    ):
+        raise HTTPException(status_code=400, detail="Invalid input data.")
 
-router = APIRouter()
+    try:
+        for field, value in data_dict.items():
+            setattr(movie, field, value)
+        await db.commit()
+    except Exception:
+        raise HTTPException(status_code=400, detail="Invalid input data.")
```

The try/except catches all exceptions and maps them to a 400. This can hide unexpected errors. Prefer validating data before assignment and only catching specific exceptions, or at least log unexpected exceptions before converting to the required 400 response. Also ensure you commit only after successful validation and consider refreshing the instance if you need to return it.